sys/log/include/log/log.h

```diff
@@ -72,11 +86,22 @@ struct log_entry_hdr {
 #define LOG_MODULE_NIMBLE_CTLR      (3)
 #define LOG_MODULE_NIMBLE_HOST      (4)
 #define LOG_MODULE_NFFS             (5)
-#define LOG_MODULE_PERUSER          (64)
+#define LOG_MODULE_MAX              (255)
+
+#define LOG_MODULE_STR(module) \
+    (LOG_MODULE_DEFAULT     == module ? "DEFAULT"     :\
+    (LOG_MODULE_OS          == module ? "OS"          :\
+    (LOG_MODULE_NEWTMGR     == module ? "NEWTMGR"     :\
+    (LOG_MODULE_NIMBLE_CTLR == module ? "NIMBLE_CTLR" :\
+    (LOG_MODULE_NIMBLE_HOST == module ? "NIMBLE_HOST" :\
+    (LOG_MODULE_NFFS        == module ? "NFFS"        :\
+     "UNKNOWN"))))))
```

Comment:
Why don't you just use a macro to string expansion on "module", which will give you the name of the define.  Also, if the condition checking remains, this should be a function, not a macro.